log/sequencer.go

```diff
@@ -207,21 +206,12 @@ func (s Sequencer) SequenceBatch(limit int, expiryFunc CurrentRootExpiredFunc) (
 	}
 
 	// Assign leaf sequence numbers and collate node updates
-	nodeMap, sequenceNumbers, err := s.sequenceLeaves(merkleTree, leaves)
+	nodeMap, leaves, err := s.sequenceLeaves(merkleTree, leaves)
```

Comment:
Maybe return into `sequencedLeaves` and restore the `len(sequencedLeaves)==len(leaves)` consistency check below?